configuration/postgres/postgres.go

```diff
@@ -0,0 +1,415 @@
+	defer conn.Release()
+	_, err = conn.Exec(ctx, channel)
+	if err != nil {
+		p.logger.Errorf("Error listening to channel:", err)
+		return
+	}
+
+	for {
+		notification, err := conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			if !(pgconn.Timeout(err) || errors.Is(ctx.Err(), context.Canceled)) {
+				p.logger.Errorf("Error waiting for notification:", err)
+			}
+			return
+		}
+		p.handleSubscribedChange(ctx, handler, notification, trigger, subscription)
+	}
+}
+
+func (p *ConfigurationStore) handleSubscribedChange(ctx context.Context, handler configuration.UpdateHandler, msg *pgconn.Notification, trigger string, subscription string) {
+	defer func() {
+		if err := recover(); err != nil {
+			p.logger.Errorf("panic in handleSubscribedChange method and recovered: %s", err)
+		}
+	}()
+	payload := make(map[string]interface{})
+	err := json.Unmarshal([]byte(msg.Payload), &payload)
+	if err != nil {
+		p.logger.Errorf("Error in UnMarshal: ", err)
+		return
+	}
+
+	var key, value, version string
+	m := make(map[string]string)
+	// trigger should encapsulate the row in "data" field in the notification
+	row := reflect.ValueOf(payload["data"])
+	if row.Kind() == reflect.Map {
+		for _, k := range row.MapKeys() {
+			v := row.MapIndex(k)
+			strKey := k.Interface().(string)
+			switch strings.ToLower(strKey) {
+			case "key":
+				key = v.Interface().(string)
+				if yes := p.isSubscribed(trigger, key, subscription); !yes {
+					p.logger.Debugf("Ignoring notification for %v", key)
+					return
+				}
+			case "value":
+				value = v.Interface().(string)
+			case "version":
+				version = v.Interface().(string)
+			case "metadata":
+				a := v.Interface().(map[string]interface{})
+				for k, v := range a {
+					m[k] = v.(string)
+				}
+			}
+		}
+	}
+	e := &configuration.UpdateEvent{
+		Items: map[string]*configuration.Item{
+			key: {
+				Value:    value,
+				Version:  version,
+				Metadata: m,
+			},
+		},
+		ID: subscription,
+	}
+	err = handler(ctx, e)
+	if err != nil {
+		p.logger.Errorf("fail to call handler to notify event for configuration update subscribe: %s", err)
+	}
+}
+
+func parseMetadata(cmetadata configuration.Metadata) (metadata, error) {
+	m := metadata{}
+
+	if val, ok := cmetadata.Properties[connectionStringKey]; ok && val != "" {
+		m.connectionString = val
+	} else {
+		return m, fmt.Errorf(ErrorMissingConnectionString)
+	}
+
+	if tbl, ok := cmetadata.Properties[configtablekey]; ok && tbl != "" {
+		if !allowedChars.MatchString(tbl) {
+			return m, fmt.Errorf("invalid table name : '%v'. non-alphanumerics are not supported", tbl)
+		}
+		if len(tbl) > maxIdentifierLength {
+			return m, fmt.Errorf(ErrorTooLongFieldLength+" - tableName : '%v'. max allowed field length is %v ", tbl, maxIdentifierLength)
+		}
+		m.configTable = tbl
+	} else {
+		return m, fmt.Errorf(ErrorMissingTableName)
+	}
+
+	// configure maxTimeout if provided
+	if mxTimeout, ok := cmetadata.Properties[connMaxIdleTimeKey]; ok && mxTimeout != "" {
```

Comment:
Should a default  value be used instead of returning error?

Reply:
done. Used the same value as pgxpool.defaultMaxConnIdleTime